pkg/handlers/generic/mutation/imageregistries/credentials/inject_test.go

```diff
@@ -436,7 +455,21 @@ var _ = Describe("Generate Image registry patches", func() {
 	for testIdx := range testDefs {
 		tt := testDefs[testIdx]
 		It(tt.Name, func() {
-			capitest.AssertGeneratePatches(GinkgoT(), patchGenerator, &tt)
+			capitest.AssertGeneratePatches(GinkgoT(), patchGenerator, &tt.PatchTestDef)
+
+			// validate an OwnerReference was added to the Secret
+			if tt.expectOwnerReferenceOnSecret {
+				client, err := helpers.TestEnv.GetK8sClientWithScheme(clientScheme)
+				gomega.Expect(err).To(gomega.BeNil())
+
+				secret := newRegistryCredentialsSecret(validSecretName, request.Namespace)
+				gomega.Expect(client.Get(
+					context.Background(),
+					ctrlclient.ObjectKeyFromObject(secret),
+					secret,
+				)).To(gomega.BeNil())
+				gomega.Expect(secret.OwnerReferences).ToNot(gomega.BeEmpty())
```

Comment:
Did you consider asserting the owner kind (and maybe name)?

Reply:
Good idea, added in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/720/commits/d7920c985fb8c764cb7e43bc26085993ee437cfd